win_etw_macros/src/lib.rs

```diff
@@ -182,6 +178,19 @@
 //!
 //! There are other tools, such as the Windows Performance Recorder, which can capture ETW events.
 //!
+//! # Using this crate for kernel-mode driver development
+//!
+//! This crate can be used when developing Windows device drivers in conjunction with the [wdk](https://crates.io/crates/wdk) crate.
+//!
+//! If you are developing for kernel-mode, you should add these dependencies to your crate:
+//! ```text
+//![dependencies]
+//!win_etw_macros = { version = "^0.1.11", default-features = false, features = "windows_drivers" }
+//!win_etw_provider = { version = "^0.1.11", default-features = false, features = "windows_drivers" }
+//! ```
+//!
+//! Then apply the `#[trace_logging_provider_kernel]` macro rather than `#[trace_logging_provider]`.
```

Comment:
Rather than having a different proc macro, I think it would be better to have a single proc macro that generated code that would work for either user-mode or kernel-mode.  If necessary, that code could use macros (not proc macros, ordinary macros) that are defined in `win_etw_provider` to handle some specialization for the environment.

I haven't skimmed the PR enough to see how different the output code is per environment, but my hope is that this can be unified.